File: src/AdventOfCode/Size.cs
// Copyright (c) Martin Costello, 2015. All rights reserved.
// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.

namespace MartinCostello.AdventOfCode
{
    /// <summary>
    /// A structure representing a size.
    /// </summary>
    public struct Size
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="Size"/> structure.
        /// </summary>
        /// <param name="width">The horizontal component of the size.</param>
        /// <param name="height">The vertical component of the size.</param>
        public Size(int width, int height)
        {
            Width = width;
            Height = height;
        }

        /// <summary>
        /// Gets or sets the vertical component of this <see cref="Size"/>.
        /// </summary>
        public int Height { get; set; }

        /// <summary>
        /// Gets or sets the horizontal component of this <see cref="Size"/>.
        /// </summary>
        public int Width { get; set; }
    }
}

File: src/AdventOfCode.Site/scripts/ts/puzzles/Size.ts
// Copyright (c) Martin Costello, 2015. All rights reserved.
// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.

export class Size {
    static readonly zero = new Size(0, 0);

    constructor(public readonly width: number, public readonly height: number) {}

    divide(value: number) {
        return new Size(this.width / value, this.height / value);
    }

    equals(other: any) {
        if (other instanceof Size) {
            return this.width === other.width && this.height === other.height;
        }
        return false;
    }
}
